File: platform/src/main/java/ru/abondin/hreasy/platform/service/VacationService.java
package ru.abondin.hreasy.platform.service;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import ru.abondin.hreasy.platform.repo.vacation.VacationRepo;
import ru.abondin.hreasy.platform.service.dto.VacationDto;
import ru.abondin.hreasy.platform.service.mapper.VacationDtoMapper;

import java.time.LocalDate;

/**
 * Service to deal with employees vacations
 */
@Service
@RequiredArgsConstructor
@Slf4j
public class VacationService {
    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class VacationFilter {
        private LocalDate endDateSince = LocalDate.of(2000, 1, 1);
    }

    private final DateTimeService dateTimeService;
    private final VacationRepo vacationRepo;
    private final VacationDtoMapper mapper;

    public Flux<VacationDto> findAll(VacationFilter filter) {
        return vacationRepo.findAll(
                filter.endDateSince
        ).map(e -> mapper.vacationToDto(e));
    }
}

File: platform/src/main/kotlin/ru/abondin/hreasy/platform/service/VacationService.kt
package ru.abondin.hreasy.platform.service

import org.springframework.stereotype.Component
import reactor.core.publisher.Flux
import ru.abondin.hreasy.platform.api.vacation.VacationDto
import ru.abondin.hreasy.platform.repo.VacationRepo
import ru.abondin.hreasy.platform.repo.VacationView
import java.time.LocalDate

@Component
class VacationService(val vacationRepo: VacationRepo, val dateTimeService: DateTimeService) {
    fun findAll(filter: VacationFilter): Flux<VacationDto> = vacationRepo.findAll(
            filter.endDateSince
    ).map { e -> vacationToDTO(e) };
}

data class VacationFilter(
        val endDateSince: LocalDate = LocalDate.of(2000, 1, 1)
)

fun vacationToDTO(entry: VacationView): VacationDto {
    return VacationDto(
            entry.id,
            entry.employee,
            employeeDisplayName(entry.employeeLastname, entry.employeeFirstname, entry.employeePatronymicName),
            entry.year,
            entry.startDate,
            entry.endDate,
            entry.notes
    );
}
